src/StructuredLogViewer/Controls/BuildControl.xaml.cs

```diff
@@ -2302,6 +2302,15 @@ private bool Invoke(BaseNode treeNode)
                         {
                             // TODO: https://github.com/KirillOsenkov/MSBuildStructuredLog/issues/392
                             evaluation = node.GetNearestParentOrSelf<ProjectEvaluation>();
+

```

Comment:
we're in BuildControl, it has a Build, and Build has `FindEvaluation(id)`, perhaps try using that? also instead of doing `node is Project` try perhaps `node.GetNearestParentOrSelf<Project>()`

Reply:
This addresses the missing "Preprocess" button when viewing a project (not in evaluation folder).
![image](https://github.com/KirillOsenkov/MSBuildStructuredLog/assets/19828377/8fc41c03-5f52-4149-be4c-e3aaf5acebd0)